examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2168 @@
+        agent_position = agent_state.position
+
+        self._metric = self._euclidean_distance(
+            previous_position, agent_position
+        )
+
+
+# -----------------------------------------------------------------------------
+# # REARRANGEMENT TASK GRIPPED OBJECT SENSOR
+# -----------------------------------------------------------------------------
+_C.TASK.GRIPPED_OBJECT_SENSOR = CN()
+_C.TASK.GRIPPED_OBJECT_SENSOR.TYPE = "GrippedObjectSensor"
+# -----------------------------------------------------------------------------
+# # REARRANGEMENT TASK ALL OBJECT POSITIONS SENSOR
+# -----------------------------------------------------------------------------
+_C.TASK.OBJECT_POSITION = CN()
+_C.TASK.OBJECT_POSITION.TYPE = "ObjectPosition"
+_C.TASK.OBJECT_POSITION.GOAL_FORMAT = "POLAR"
+_C.TASK.OBJECT_POSITION.DIMENSIONALITY = 2
+# -----------------------------------------------------------------------------
+# # REARRANGEMENT TASK ALL OBJECT GOALS SENSOR
+# -----------------------------------------------------------------------------
+_C.TASK.OBJECT_GOAL = CN()
+_C.TASK.OBJECT_GOAL.TYPE = "ObjectGoal"
+_C.TASK.OBJECT_GOAL.GOAL_FORMAT = "POLAR"
+_C.TASK.OBJECT_GOAL.DIMENSIONALITY = 2
+# -----------------------------------------------------------------------------
+# # OBJECT_DISTANCE_TO_GOAL MEASUREMENT
+# -----------------------------------------------------------------------------
+_C.TASK.OBJECT_TO_GOAL_DISTANCE = CN()
+_C.TASK.OBJECT_TO_GOAL_DISTANCE.TYPE = "ObjectToGoalDistance"
+# -----------------------------------------------------------------------------
+# # OBJECT_DISTANCE_FROM_AGENT MEASUREMENT
+# -----------------------------------------------------------------------------
+_C.TASK.AGENT_TO_OBJECT_DISTANCE = CN()
+_C.TASK.AGENT_TO_OBJECT_DISTANCE.TYPE = "AgentToObjectDistance"
+
+from habitat.config.default import CN, Config
+
+# %%
+# @title Define `RearrangementTask` by extending `NavigationTask`
+from habitat.tasks.nav.nav import NavigationTask, merge_sim_episode_config
+
+
+def merge_sim_episode_with_object_config(
+    sim_config: Config, episode: Type[Episode]
+) -> Any:
+    sim_config = merge_sim_episode_config(sim_config, episode)
+    sim_config.defrost()
+    sim_config.objects = [episode.objects.__dict__]
+    sim_config.freeze()
+
+    return sim_config
+
+
+@registry.register_task(name="RearrangementTask-v0")
+class RearrangementTask(NavigationTask):
+    r"""Embodied Rearrangement Task
+    Goal: An agent must place objects at their corresponding goal position.
+    """
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+    def overwrite_sim_config(self, sim_config, episode):
+        return merge_sim_episode_with_object_config(sim_config, episode)
+
+
+# %% [markdown]
+# ## Implement a hard-coded and an RL agent
+#
+#
+
+# %%
+# @title Load the `RearrangementTask` in Habitat-Lab and run a hard-coded agent
+import habitat
+
+config_file = """
```

Comment:
You could parameterize these config files with format strings and have the user specify the values with sliders / other widgets if you wanted to. Just like we do in the interactivity tutorial. 

You can also contruct these configs as recursive python directionaries. They are yamls after all. You can generate how the dictionaries by running yaml.loads

Reply:
Yeah, I was wondering about this. Initially I thought, I will just write it to a yaml file to tell the user that look you can just change the config YAML file and load it back. But this looks more ugly. Will change it .